Formatting fails with mixed comments

When debugging I will mix block and line comments to quickly comment out sections of code, so I can move from this:
```go 
#include "textflag.h"

TEXT ·FailsFormatting(SB), NOSPLIT, $0
        /*/
        MOVQ fail_spectacularly+0(FP), SP //*/
        RET
```

to this, by deleting a single `*`:
```go 
#include "textflag.h"

TEXT ·FailsFormatting(SB), NOSPLIT, $0
        //
        MOVQ fail_spectacularly+0(FP), SP //*/
        RET
```

I'm getting a panic from asmfmt when it sees this pattern.  I'm not sure the preferred action for the majority here, but I would like it to be left unchanged when it's within a line comment like this.  

Minimal test:
```go
#include "textflag.h"

TEXT ·FailsFormatting(SB), NOSPLIT, $0
        RET //*/
```

Panic output:
```
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
panic(0x520100, 0xc82000e080)
       /src/runtime/panic.go:481 +0x3e6
github.com/klauspost/asmfmt.(*fstate).addLine(0xc82003baf8, 0xc82008b8c7, 0x14, 0x739, 0x0, 0x0)
        /src/github.com/klauspost/asmfmt/asmfmt.go:153 +0x1c1c
github.com/klauspost/asmfmt.Format(0x7f98752d5410, 0xc82004c070, 0x0, 0x0, 0x0, 0x0, 0x0)
        /src/github.com/klauspost/asmfmt/asmfmt.go:32 +0x327
main.processFile(0x7ffcf3bd2700, 0x13, 0x7f98752d53e8, 0xc820028028, 0x7f98752d53c0, 0xc820028010, 0x0, 0x0, 0x0)
        /src/github.com/klauspost/asmfmt/cmd/asmfmt/main.go:79 +0x248
main.gofmtMain()
        /src/github.com/klauspost/asmfmt/cmd/asmfmt/main.go:170 +0x763
main.main()
        /src/github.com/klauspost/asmfmt/cmd/asmfmt/main.go:130 +0x18
```

--- a/asmfmt.go
+++ b/asmfmt.go
@@ -67,7 +67,6 @@ func (f *fstate) addLine(b []byte) error {
 		return fmt.Errorf("zero (0) byte in input. file is unlikely an assembler file")
 	}
 	s := string(b)
-	s = strings.TrimSpace(s)
 	// Inside block comment
 	if f.insideBlock {
 		defer func() {
@@ -95,7 +94,8 @@ func (f *fstate) addLine(b []byte) error {
 			}
 		} else {
 			// Insert a space on lines that begin with '*'
-			if strings.HasPrefix(s, "*") {
+			if strings.HasPrefix(strings.TrimSpace(s), "*") {
+				s = strings.TrimSpace(s)
 				f.out.WriteByte(' ')
 				f.lastStar = true
 			} else {
@@ -105,6 +105,7 @@ func (f *fstate) addLine(b []byte) error {
 			return nil
 		}
 	}
+	s = strings.TrimSpace(s)
 
 	// Comment is the the only line content.
 	if strings.HasPrefix(s, "//") {
@@ -141,6 +142,15 @@ func (f *fstate) addLine(b []byte) error {
 	if strings.Contains(s, "/*") && !strings.HasSuffix(s, `\`) {
 		starts := strings.Index(s, "/*")
 		ends := strings.Index(s, "*/")
+		lineComment := strings.Index(s, "//")
+		if lineComment >= 0 {
+			if lineComment < starts {
+				goto exitcomm
+			}
+			if lineComment < ends && !f.insideBlock {
+				goto exitcomm
+			}
+		}
 		pre := s[:starts]
 		pre = strings.TrimSpace(pre)
 

--- a/asmfmt.go
+++ b/asmfmt.go
@@ -74,12 +74,11 @@ func (f *fstate) addLine(b []byte) error {
 		}()
 		if strings.Contains(s, "*/") {
 			ends := strings.Index(s, "*/")
-			end := strings.TrimSpace(s[:ends])
-			if f.lastStar {
-				end = end + " */"
-			} else {
-				end = end + "*/"
+			end := s[:ends]
+			if strings.HasPrefix(strings.TrimSpace(s), "*") && f.lastStar {
+				end = strings.TrimSpace(end) + " "
 			}
+			end = end + "*/"
 			f.insideBlock = false
 			s = strings.TrimSpace(s[ends+2:])
 			if strings.HasSuffix(s, "\\") {

--- a/testdata/mod_math_amd64.golden
+++ b/testdata/mod_math_amd64.golden
@@ -1,7 +1,7 @@
 /*
 Quicker way of loading MOD_P ?
-MOVL    $-1, R8
-NEGQ    R8
+        MOVL    $-1, R8
+        NEGQ    R8
 */
 #define MOD_P 0xFFFFFFFF00000001
 

--- a/testdata/nested_comment.golden
+++ b/testdata/nested_comment.golden
@@ -5,14 +5,14 @@ TEXT ·FailsFormatting(SB), NOSPLIT, $0
 
 TEXT ·FailsFormatting(SB), NOSPLIT, $0
 /*
-RET // */
+        RET //*/
 
 TEXT ·FailsFormatting(SB), NOSPLIT, $0
 	RET // Some longer comment*/
 
 TEXT ·FailsFormatting(SB), NOSPLIT, $0
 /*
-RET // Some longer comment */
+        RET // Some longer comment*/
 
 TEXT ·FailsFormatting(SB), NOSPLIT, $0
 	/*        RET // */

--- a/testdata/nested_comment.golden
+++ b/testdata/nested_comment.golden
@@ -1,0 +1,28 @@
+#include "textflag.h"
+
+TEXT ·FailsFormatting(SB), NOSPLIT, $0
+	RET // */
+
+TEXT ·FailsFormatting(SB), NOSPLIT, $0
+/*
+RET // */
+
+TEXT ·FailsFormatting(SB), NOSPLIT, $0
+	RET // Some longer comment*/
+
+TEXT ·FailsFormatting(SB), NOSPLIT, $0
+/*
+RET // Some longer comment */
+
+TEXT ·FailsFormatting(SB), NOSPLIT, $0
+	/*        RET // */
+
+/*
+	TESTL	BX, BX
+	JEQ	move_0
+	CMPL	BX, $2
+	JBE	move_1or2
+	CMPL	BX, $4
+	JBE	move_3or4
+	CMPL	BX, $8
+*/

--- a/testdata/nested_comment.in
+++ b/testdata/nested_comment.in
@@ -1,0 +1,30 @@
+#include "textflag.h"
+
+TEXT ·FailsFormatting(SB), NOSPLIT, $0
+        RET //*/
+
+
+TEXT ·FailsFormatting(SB), NOSPLIT, $0
+/*
+        RET //*/
+
+TEXT ·FailsFormatting(SB), NOSPLIT, $0
+        RET // Some longer comment*/
+
+
+TEXT ·FailsFormatting(SB), NOSPLIT, $0
+/*
+        RET // Some longer comment*/
+
+TEXT ·FailsFormatting(SB), NOSPLIT, $0
+/*        RET //*/
+
+/*
+	TESTL	BX, BX
+	JEQ	move_0
+	CMPL	BX, $2
+	JBE	move_1or2
+	CMPL	BX, $4
+	JBE	move_3or4
+	CMPL	BX, $8
+*/

